### ISSUE TITLE
Failed build messes up printing

### ISSUE THREAD

**Issue by pv** [spacetelescope/asv#209]
It appears sometimes a build failure results to clobbered print output:
```
git clone https://github.com/pv/scipy-bench.git scipy-bench
cd scipy-bench
git checkout origin/param
./run.py run -n master^! # to make benchmark discovery work
./run.py run -n e80cb04^^..e80cb04
```
results to the following output from `asv`: https://gist.github.com/pv/7b73f66f404f58703ba7
```
... huge  build log output ...
...
 error: Command "x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall [...] special/_cephesmodule.o" failed with exit status 1
 
[ 0.00%] ·· Skipping py2.7-Cython0.21.1-Tempita0.5.2-numpy1.8.2-six
[ 1.09%] ··· Benchmark fftpack_basic.RFft.time_rfft skipped
[ 2.17%] ··· Benchmark sparse.Construction.time_construction skipped 
...
[ 48.91%] ··· Benchmark fftpack_basic.Fftn.time_fftn skipped
[ 50.00%] ··· Benchmark sparse_linalg_expm.ExpmMultiply.time_expm_multiply skipped... failed failed failed failed failed failed failed failed failed 
```
The first build fails (as it should, buggy commit), but the output from the second build command becomes strange --- it's as if the asv logger stops producing output, after printing out the skipped benchmark messages. The only thing that is printed are apparently the `log.add` stuff which goes directly to stdout.

Didn't manage to reproduce this yet in a simpler case and smaller repository (sorry)...

**Comment by pv** [comment]
This was actually tested with gh-203 present, not against master

**Comment by pv** [comment]
Ok, this occurred only with gh-203, fixed by pv@044b8d6f9c
